src/consumer/index.js

```diff
@@ -153,7 +109,7 @@ module.exports = ({
   }
 
   /** @type {import("../../types").Consumer["stop"]} */
-  const stop = async () => {
+  const stop = sharedPromiseTo(async () => {
```

Comment:
Not that it hurts to make this a shared promise, but is there a particular reason for it to be?

Reply:
`process.on('SIGTERM', () => consumer.disconnect())`

Send SIGTERM signal twice and consumerGroup.leave() will reopen the connection, which will increase the graceful shutdown time.